macOS CI failing

<!--
**Bug reports will only be accepted against the current tip of the master git branch. Anything else will be rejected. Questions must be posted in the discussion board at <https://github.com/aircrack-ng/aircrack-ng/discussions>.**

For any security-related issues such as exploits, buffer overflows, and such, refer to our security policy: <https://github.com/aircrack-ng/aircrack-ng/security/policy>

Please read the following before reporting the issue:

- Use the search function to check if the bug you're about to post isn't a duplicate. If an existing bug is open and you have new information, update it. If a bug exists and is closed, reopen it and add useful information to it.
- Unless bugs are linked, one bug per ticket.
- Don't worry about any of the items on the right panel (Assignee, Labels or Milestone), we will take care of them.

The following must be taken to the GitHub Discussions first, as in these cases, it often turns out to be a technique issue or the issue is out of our control:

- Not receiving any packets with airodump-ng or any other tool.
- Can't crack a network.
- Wireless card doesn't work.
- Channel -1.

Make sure to check our documentation at <https://aircrack-ng.org/doku.php> as well as our fairly comprehensive FAQ at <https://aircrack-ng.org/doku.php?id=faq>.

**REMOVE ALL OF THE ABOVE TEXT**
-->

# Defect Report

## Issue type

<!--
Please select from one of the following software defect descriptions. Pick the one(s) that best reflects the issue you are experiencing. Erase the others.
-->

- [X] Defect - Compilation or `make check` issue - Attach `config.log` and provide relevant system information such as `lscpu` - Make sure dependencies are installed

## System information

<!--
On Linux, OS version can often be found in /etc/os-release. On Debian-based you may use `lsb_release -i` for "OS Name", `lsb_release -r` for "OS version".

Copy and paste the output of `uname -a` in Kernel version.

CPU information can be gathered with `lscpu`, or `cat /proc/cpuinfo`

Wireless card and chipset information is displayed with `airmon-ng`. If it doesn't display anything, add information from `lspci`, `lsusb`, and `sudo lshw -c network -sanitize`. The wireless adapter can also be found in dmesg.
-->

- OS Name: macOS
- CPU: i386

## Aircrack-ng version

<!--
For the aircrack-ng version, run `aircrack-ng | head`, it will be displayed right after 'Aircrack-ng'.
When compiled from git, the short SHAsum (7 characters long) will be shown right after the version number.

Don't say 'Latest' for the version, this is not helpful as it changes over time.
-->

- Version: 1.7
- Commit Revision hash: https://github.com/aircrack-ng/aircrack-ng/commit/d0c32c8bb8bc571d542f3f31d92edd60a29d7da1

## Airmon-ng debug information

N/A

<!--
If the issue is related to packet capture or injection, run `airmon-ng --debug` and post the output here.

Please make sure to enclose with backticks for readability.
-->

## Defect

### Details

The action fails with `clang`, see [this job](https://github.com/aircrack-ng/aircrack-ng/actions/runs/4240756537/jobs/7370146024) for example:

```
   ../test/unit/test-circular-buffer.c:102:2: error: passing 'uint8_t[17]' to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Werror,-Wpointer-sign]
          assert_string_equal("a1234567b1234567", buffer);
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /usr/local/include/cmocka.h:1363:31: note: expanded from macro 'assert_string_equal'
      _assert_string_equal((a), (b), __FILE__, __LINE__)
                                ^~~
  /usr/local/include/cmocka.h:2352:68: note: passing argument to parameter 'b' here
  void _assert_string_equal(const char * const a, const char * const b,
                                                                     ^
  1 error generated.
  make[2]: *** [test/unit/circular_buffer-test-circular-buffer.o] Error 1
```

<!--
Summarize the issue in a paragraph or two.

If you need to provide output of the different tools, or commands, file content, etc, use the backticks to enclose the text.

Only post pictures if the issues when it cannot be avoided.
-->

### How to reproduce the issue

Commit to the repository.

<!--
A clear and concise list of steps describing how to reproduce the issue, relevant commands and errors/outputs. Such as: What command(s) did you run? What was displayed/happened? What did you expect to happen or be displayed?

Do not use pastebin-type links as they expire, rendering the bug report unactionable. Avoid using images whenever possible. If the content (output or error) is text and can be copy/pasted, enclose it with backticks for readability. If it is large, put it in a file and attach it to the ticket.

GitHub markdown guide: https://guides.github.com/features/mastering-markdown/

Last, but not least, if you aren't familiar with bug reporting, [this](https://www.chiark.greenend.org.uk/~sgtatham/bugs.html) is an excellent read to help you describe bugs accurately.
-->

## Related issues

<!--
Link to any related issue from within the project or outside. Such as GitHub, other bug trackers, relevant GitHub Discussion post, relevant documentation.
-->

N/A

--- a/test/unit/test-circular-buffer.c
+++ b/test/unit/test-circular-buffer.c
@@ -99,7 +99,7 @@ static void test_cbuf_multibyte_compare_buffer(void ** state)
 	assert_int_equal(nb_elements, circular_buffer_size(cbuf));
 	assert_false(circular_buffer_is_empty(cbuf));
 	assert_true(circular_buffer_is_full(cbuf));
-	assert_string_equal("a1234567b1234567", buffer);
+	assert_memory_equal("a1234567b1234567", buffer, size);
 
 	// END
 #undef size

